src/engine/Minus.cpp

```diff
@@ -0,0 +1,230 @@
+  // disjoint (mu' defines no solution for any variables for which mu defines a
+  // solution).
+
+  // The output is always the same size as the left input
+  constexpr int OUT_WIDTH = A_WIDTH;
+
+  // check for trivial cases
+  if (dynA.size() == 0) {
+    return;
+  }
+
+  if (dynB.size() == 0 || joinColumns.size() == 0) {
+    // B is the empty set of solution mappings, so the result is A
+    // Copy a into the result, allowing for optimizations for small width by
+    // using the templated width types.
+    // TODO<joka921> Check, if this direct assignment is efficient
+    // (It should be, because it internally copies a std::vector<char> via
+    // memcpy
+    *dynResult = dynA;
+    return;
+  }
+
+  IdTableView<A_WIDTH> a = dynA.asStaticView<A_WIDTH>();
+  IdTableView<B_WIDTH> b = dynB.asStaticView<B_WIDTH>();
+  IdTableStatic<OUT_WIDTH> result = dynResult->moveToStatic<OUT_WIDTH>();
+
+  std::vector<size_t> rightToLeftCols(b.cols(),
+                                      std::numeric_limits<size_t>::max());
+  for (const auto& jc : joinColumns) {
+    rightToLeftCols[jc[1]] = jc[0];
+  }
+
+  /**
+   * @brief A function to copy a row from a to the end of result.
+   * @param ia The index of the row in a.
+   */
+  auto writeResult = [&result, &a](size_t ia) {
+    result.emplace_back();
+    size_t backIdx = result.size() - 1;
+    for (size_t col = 0; col < a.cols(); col++) {
+      result(backIdx, col) = a(ia, col);
+    }
+  };
+
+  size_t ia = 0, ib = 0;
+  while (ia < a.size() && ib < b.size()) {
+    // Join columns 0 are the primary sort columns
+    while (a(ia, joinColumns[0][0]) < b(ib, joinColumns[0][1])) {
+      // Write a result
+      writeResult(ia);
+      ia++;
+      if (ia >= a.size()) {
+        goto finish;
+      }
+    }
+    while (b(ib, joinColumns[0][1]) < a(ia, joinColumns[0][0])) {
+      ib++;
+      if (ib >= b.size()) {
+        goto finish;
+      }
+    }
+
+    while (b(ib, joinColumns[0][1]) == a(ia, joinColumns[0][0])) {
+      // check if the rest of the join columns also match
+      RowComparison rowEq = isRowEqSkipFirst(a, b, ia, ib, joinColumns);
+      switch (rowEq) {
+        case RowComparison::EQUAL: {
+          ia++;
+          if (ia >= a.size()) {
+            goto finish;
+          }
+        } break;
+        case RowComparison::LEFT_SMALLER: {
+          // ib does not discard ia, and there can not be another ib that
+          // would discard ia.
+          writeResult(ia);
+          ia++;
+          if (ia >= a.size()) {
+            goto finish;
+          }
+        } break;
+        case RowComparison::RIGHT_SMALLER: {
+          ib++;
+          if (ib >= b.size()) {
+            goto finish;
+          }
+        } break;
+        default:
+          AD_CHECK(false);
+      }
+    }
+  }
+finish:
+  result.reserve(result.size() + (a.size() - ia));
+  while (ia < a.size()) {
+    writeResult(ia);
+    ia++;
+  }
```

Comment:
Doesn't the `IdTable` support `std::insert()` or something like this?
If not, it probably should.